client/src/components/_common/TextCopyField/TextCopyField.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import PropTypes from 'prop-types';
+import { Button } from 'reactstrap';
+import Icon from '../Icon';
+import './TextCopyField.module.scss';
+
+const TextCopyField = ({ value, placeholder }) => {
+  const onChange = event => {
+    // Swallow keyboard events on the Input control, but
+    // still allow selecting the text. readOnly property of
+    // Input is not adequate for this purpose because it
+    // prevents text selection
+    event.preventDefault();
+  };
+  const onCopy = onChange;
+  return (
+    <div className="input-group" styleName="root">
+      <div className="input-group-prepend">
+        <CopyToClipboard text={value}>
+          <Button styleName="copy-button" onClick={event => onCopy(event)}>
+            <Icon name="link" styleName="button__icon" />
+            <span styleName="button__text">Copy</span>
+          </Button>
+        </CopyToClipboard>
+      </div>
+      <input
+        type="text"
+        onChange={onChange}
+        value={value || ''}
```

Comment:
__Notice__: I was specifically asked to change very similar code to use `defaultProps`, instead of ` || ''`, by two coworkers: [request #1](https://github.com/TACC/Frontera-Portal/pull/262#discussion_r533042675), [request #2](https://github.com/TACC/Frontera-Portal/pull/262#discussion_r535494227), [commit](https://github.com/TACC/Frontera-Portal/pull/262/commits/6f77ee540caa93500b09048223b2cbb7c096f475).